scaleway/resource_instance_server_test.go

```diff
@@ -428,6 +428,25 @@ func TestAccScalewayInstanceServerIpv6(t *testing.T) {
 	})
 }
 
+func TestAccScalewayInstanceServerWithReservedIP(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckScalewayInstanceServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckScalewayInstanceServerConfigWithReservedIP,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceServerExists("scaleway_instance_server.base"),
+					testAccCheckScalewayInstanceIPExists("scaleway_instance_ip.ip"),
+					resource.TestCheckResourceAttrPair("scaleway_instance_ip.ip", "address", "scaleway_instance_server.base", "public_ip"),
+					resource.TestCheckResourceAttrPair("scaleway_instance_ip.ip", "id", "scaleway_instance_server.base", "ip_id"),
+				),
+			},
```

Comment:
Add a step where you change an IP by another.  This is the point not cover by your code see comment above

Reply:
:+1: